Content.Shared/_RMC14/Atmos/SharedRMCFlammableSystem.cs

```diff
@@ -119,6 +120,26 @@ public override void Initialize()
 
     private void OnIgniteOnProjectileHit(Entity<IgniteOnProjectileHitComponent> ent, ref ProjectileHitEvent args)
     {
+        // Check for complete ignition immunity first
+        if (TryComp<RMCImmuneToIgnitionComponent>(args.Target, out var ignitionImmunity))
+        {
+            // If entity has complete ignition immunity, check if this fire can bypass it
+            if (ignitionImmunity.BypassWhitelist == null)
+            {
+                // Complete immunity - no fires can ignite this entity
+                return;
+            }
+            else
+            {
+                // Check if this fire is on the bypass whitelist
+                if (!_entityWhitelist.IsWhitelistPass(ignitionImmunity.BypassWhitelist, ent))
+                {
+                    // Fire is not on bypass whitelist, cannot ignite
+                    return;
+                }
+            }
+        }
+
```

Comment:
Make a method out of this like CanBeIgnited so you can reuse it elsewhere in this file